backend/typescript/services/implementations/camperService.ts

```diff
@@ -12,7 +12,23 @@ class CamperService implements ICamperService {
   async createCamper(camper: CreateCamperDTO): Promise<CamperDTO> {
     let newCamper: Camper;
     let existingCamp: Camp | null;
+
     try {
+      existingCamp = await MgCamp.findById(camper.camp);
+
+      if (existingCamp) {
+        const existingCampObj = existingCamp.toObject();
+
+        if (
+          existingCamp &&
+          existingCamp.campers.length >= existingCampObj.capacity
+        ) {
+          throw new Error(`Error: camp is full`);
```

Comment:
very nitpicky thing: the error message could be slightly more descriptive, like can print out the camp capacity and the length of campers